src/components/Footer/Footer.stories.tsx

```diff
@@ -0,0 +1,220 @@
+import { Meta, Story } from '@storybook/react/types-6-0';
+import { Footer } from '.';
+import { BsDribbble, BsFacebook, BsGithub, BsInstagram, BsTwitter } from 'react-icons/bs';
+
+export default {
+  title: 'Components/Footer',
+  component: Footer,
+} as Meta;
+
+const Template: Story = ({ children, className }) => <Footer className={className}>{children}</Footer>;
+
+export const DefaultFooter = Template.bind({});
+DefaultFooter.storyName = 'Default';
+DefaultFooter.args = {
+  children: (
+    <>
+      <Footer.Copyright href="#" createby="Flowbite™" year={2022} />
+      <Footer.LinkGroup className="mt-3 flex-wrap items-center text-sm sm:mt-0">
+        <Footer.Link href="#">About</Footer.Link>
+        <Footer.Link href="#">Privacy Policy</Footer.Link>
+        <Footer.Link href="#">Licensing</Footer.Link>
+        <Footer.Link href="#">Contact</Footer.Link>
+      </Footer.LinkGroup>
+    </>
+  ),
+};
+
+export const WithLogoFooter = Template.bind({});
+WithLogoFooter.storyName = 'With Logo';
+WithLogoFooter.args = {
+  className: 'flex flex-col',
+  children: (
+    <>
+      <div className="w-full justify-between sm:flex sm:items-center sm:justify-between">
+        <Footer.Brand
+          href="https://flowbite.com"
+          src="https://flowbite.com/docs/images/logo.svg"
+          alt="Flowbite Logo"
+          name="Flowbite"
+        />
+
+        <Footer.LinkGroup className="mt-3 flex-wrap items-center text-sm sm:mt-0">
+          <Footer.Link href="#">About</Footer.Link>
+          <Footer.Link href="#">Privacy Policy</Footer.Link>
+          <Footer.Link href="#">Licensing</Footer.Link>
+          <Footer.Link href="#">Contact</Footer.Link>
+        </Footer.LinkGroup>
+      </div>
+      <hr className="my-6 w-full border-gray-200 p-1 dark:border-gray-700 sm:mx-auto lg:my-8" />
+      <Footer.Copyright href="#" createby="Flowbite™" year={2022} />
+    </>
+  ),
+};
+
+export const WithSocialMediaFooter = Template.bind({});
+WithSocialMediaFooter.storyName = 'Social Media Icons';
+WithSocialMediaFooter.args = {
+  className: 'flex flex-col',
+  children: (
+    <>
+      <div className="grid w-full justify-between sm:flex sm:justify-between md:flex md:grid-cols-1">
+        <div>
+          <Footer.Brand
+            href="https://flowbite.com"
+            src="https://flowbite.com/docs/images/logo.svg"
+            alt="Flowbite Logo"
+            name="Flowbite"
+          />
+        </div>
+        <div className="grid grid-cols-2 gap-8 sm:mt-4 sm:grid-cols-3 sm:gap-6">
+          <Footer.Col title="About">
```

Comment:
I'm not sure about this `Footer.Col` component. I would prefer to have it in pure jsx inside the `Footer`. What do you think @bacali95 @tulup-conner ?

Reply:
@rluders I picture:

```
<Footer>
  <Footer.Col>
    <Footer.Heading as="h3">Company</Footer.Heading>
    <Footer.LinkGroup>
      <Footer.Link href="#">...</Footer.Link>
      ...
    </Footer.LinkGroup>
    ...
  </Footer.Col>
  ...
</Footer>
```

I think `Footer.Col` is preferred over having the user add the `div class="flex-col"` themselves. But the `title` should not be rolled in.

Thoughts:

- Heading levels should not be enforced by a component library; should be able to pass `as="h3"`, etc.
- Can `Footer.IconContent` just be `Footer.Icon`?
- `Footer.Copyright#createby` is a confusing name. I think this should just be replaced by using `children`

Reply:
I think that to have just a component to add a `<div className="flex flex-col">...</div>` is too much. I think that to have just a component to add a `<div className="flex flex-col">...</div>` is too much. However, I may agree that would be nice to have some `grid` or `flex` components not just for the Footer.

```jsx
<Grid cols={4} gap={4}>
    <Grid.Col>...</Grid.Col>
    <Grid.Col>...</Grid.Col>
    <Grid.Col>...</Grid.Col>
    <Grid.Col>...</Grid.Col>
<Grid>
```

If we have something like this, this concept could/should replace the `<Footer.Col>`.
I liked the idea of the `Heading` component, but I also think that it should be part of some other components group.

- `Footer.Icon` agreed
- `Footer.Copyright`, I agree with the naming, but I think that the component looks very nice. Maybe it could give both possibility? To support children or params?

Reply:
@bacali95 @tulup-conner kind ping to continue the discussion here. :)

Reply:
@tulup-conner  and @rluders thanks for your relevant feedback, based on these I made some changes.

Reply:
Heading and Layout components should/will be handled in separated PRs.